src/core/runtime/interrupt_signal.cpp

```diff
@@ -144,8 +144,14 @@ hsa_signal_value_t InterruptSignal::WaitRelaxed(
 
   uint32_t prior = waiting_++;
   MAKE_SCOPE_GUARD([&]() { waiting_--; });
-  // Allow only the first waiter to sleep (temporary, known to be bad).
-  if (prior != 0) wait_hint = HSA_WAIT_STATE_ACTIVE;
+
+  uint64_t event_age = 1;
+
+  if (!core::Runtime::runtime_singleton_->KfdVersion().supports_event_age) {
+      event_age = 0;
+      // Allow only the first waiter to sleep (temporary, known to be bad).
```

Comment:
The "(temporary ...)" part of the comment can be removed, or replaced with an explanation why this is needed when event age is not supported by KFD.